components/layout/layout_thread.rs

```diff
@@ -908,9 +917,40 @@ impl LayoutThread {
                 debug!("Layout done!");
 
                 self.epoch.next();
-                self.paint_chan
-                    .send(LayoutToPaintMsg::PaintInit(self.epoch, paint_layer))
-                    .unwrap();
+
+                if opts::get().use_webrender {
+                    let api = self.webrender_api.as_ref().unwrap();
+                    // TODO: Avoid the temporary conversion and build webrender sc/dl directly!
+                    let Epoch(epoch_number) = self.epoch;
+                    let epoch = webrender_traits::Epoch(epoch_number);
+                    let pipeline_id = self.id.to_webrender();
+
+                    // TODO(gw) For now only create a root scrolling layer!
```

Comment:
Iframes work - there is a root scrolling layer per pipeline. What doesn't work yet is overflow: scroll within a normal page - I was going to wait until mrobinson's display list changes for that, since we have a heap of hacks at the moment related to creating a fake stacking context to handle this case in Servo.